Automattic/wp-calypso
# A4a: Streamline sign-up flow - Redirect logged out users to WPCOM

Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/352
Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/353

## Proposed Changes

* This PR adds the logic necessary to redirect logged-out users to WPCOM based on their email:
   * Existing WPCOM users are redirected to the login page
   * Non-existing WPCOM users are redirected to WPCOM's signup page

## Testing Instructions

**NOTE:** This doesn't work for `automattic.com` emails, please use another email domain for testing.

* Apply the diff D146625-code to your sandbox
* Point `public-api.wordpress.com` in your hosts file to your sandbox
* Pull this branch and restart a8c-for-agencies ( `yarn start-a8c-for-agencies` )
* Make sure you are logged out of WPCOM (using a private browser window helps here)
* Go to the signup page (http://agencies.localhost:3000/signup)
* Fill in the form using an existing WPCOM email
  * Verify that you are redirected to the WPCOM login page
  * Log in and authorize A4A
  * Verify that after authorization you are redirected to `/signup/resume` and your form is displayed as a JSON string on the page
* Fill in the form using a non-existing WPCOM email
  * Verify that you are redirected to the WPCOM signup page
  * Create a new account and authorize A4A
  * Verify that after authorization you are redirected to `/signup/resume` and your form is displayed as a JSON string on the page

## Pre-merge Checklist
